numpy/core/src/umath/override.c

```diff
@@ -29,7 +29,7 @@ normalize_signature_keyword(PyObject *normal_kwds)
                             "cannot specify both 'sig' and 'signature'");
             return -1;
         }
-        Py_INCREF(obj);
+        /* No INCREF needed, as PyDict_SetItem takes its own */
```

Comment:
By "takes its own", you mean "borrows the reference"?

Reply:
No, it takes its own reference; that is why no `INCREF` is needed. This in contrast to `PyList_SetItem`. Should I change the comment to say "does not steal it"?

Reply:
I like the alternative text better, and think that it would be even easier to parse for humans as a positive, rather than a negative, statement, i.e. "increases the reference count."